pkg/container/network.go

```diff
@@ -184,28 +204,48 @@ func takeAddress(iface *net.Interface) (*net.IPNet, bool, error) {
 	return nil, false, fmt.Errorf("interface %s has no valid addresses", iface.Name)
 }
 
-func createTAPAdapter(tapName string) error {
-	if _, err := util.ExecuteCommand("ip", "tuntap", "add", "mode", "tap", tapName); err != nil {
+func createTAPAdapter(handle *netlink.Handle, tapName string) error {
+	la := netlink.NewLinkAttrs()
+	la.Name = tapName
+	tuntap := &netlink.Tuntap{
+		LinkAttrs: la,
+		Mode:      netlink.TUNTAP_MODE_TAP,
+	}
+	err := netlink.LinkAdd(tuntap)
+	if err != nil {
 		return err
 	}
-
-	return setLinkUp(tapName)
+	return setLinkUp(handle, tapName)
```

Comment:
We should likely call `handle.LinkSetUp(tuntap)` here and get rid of `setLinkUp` entirely